### PR TITLE
Enable debugging of WebView on Android

### PR DESCRIPTION
**Pull request by rynkowsg**
Problem:   Debugging of WebView is disabled.
Solution:  Set proper flag for Android WebView to enable debugging.

Reference:
https://developer.chrome.com/devtools/docs/remote-debugging#configure-webview

### CODE REVIEW

File: cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java [hunk 1]
```diff
@@ -66,6 +68,13 @@ public Cocos2dxWebView(Context context, int viewTag) {
 
         this.setWebViewClient(new Cocos2dxWebViewClient());
         this.setWebChromeClient(new WebChromeClient());
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
```
Comment (minggo) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java:71]: Why need to do this check?

Comment (rynkowsg) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java:71]: `setWebContentsDebuggingEnabled` has been added in Android's API level 19 (KITKAT). There is no sense to call it on platforms with API version < 19. 

As you can see in the [build log](http://45.56.80.45:8080/job/ccs-pr/node=android/4351/console), there are two errors:

```
[javac] Compiling 28 source files to /android_slave/workspace/ccs-pr/node/android/cocos/platform/android/java/bin/classes
    [javac] /android_slave/workspace/ccs-pr/node/android/cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java:71: error: cannot find symbol
    [javac]         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
    [javac]                                                         ^
    [javac]   symbol:   variable KITKAT
    [javac]   location: class VERSION_CODES
    [javac] /android_slave/workspace/ccs-pr/node/android/cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java:75: error: cannot find symbol
    [javac]                 WebView.setWebContentsDebuggingEnabled(true);
    [javac]                        ^
    [javac]   symbol:   method setWebContentsDebuggingEnabled(boolean)
    [javac]   location: class WebView
    [javac] Note: /android_slave/workspace/ccs-pr/node/android/cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxDownloader.java uses unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 2 errors
```

Both `Build.VERSION_CODES.KITKAT` and `setWebContentsDebuggingEnabled(boolean)` don't exist in the version of target API. In the [build log](http://45.56.80.45:8080/job/ccs-pr/node=android/4351/console) we can find that Ant is using API 10 (Android 2.3.3). 

```
-build-setup:
[getbuildtools] Using latest Build Tools: 23.0.0 rc2
     [echo] Resolving Build Target for cocos2dxandroid...
[gettarget] Project Target:   Android 2.3.3
[gettarget] API level:        10
     [echo] ----------
     [echo] Creating output directories if needed...
    [mkdir] Created dir: /android_slave/workspace/ccs-pr/node/android/cocos/platform/android/java/bin
    [mkdir] Created dir: /android_slave/workspace/ccs-pr/node/android/cocos/platform/android/java/bin/res
    [mkdir] Created dir: /android_slave/workspace/ccs-pr/node/android/cocos/platform/android/java/bin/rsObj
    [mkdir] Created dir: /android_slave/workspace/ccs-pr/node/android/cocos/platform/android/java/bin/rsLibs
    [mkdir] Created dir: /android_slave/workspace/ccs-pr/node/android/cocos/platform/android/java/gen
    [mkdir] Created dir: /android_slave/workspace/ccs-pr/node/android/cocos/platform/android/java/bin/classes
    [mkdir] Created dir: /android_slave/workspace/ccs-pr/node/android/cocos/platform/android/java/bin/dexedLibs
     [echo] ----------
     [echo] Resolving Dependencies for cocos2dxandroid...
[dependency] Library dependencies:
[dependency] No Libraries
[dependency] 
[dependency] ------------------
[dependency] API<=15: Adding annotations.jar to the classpath.
```

I checked the configuration of that test (CppTest). It is in `tests/cpp-tests/proj.android/project.properties` and 
`cocos/platform/android/java/project.properties`. `target` should be set there to `android-22`
If you run test using gradle everything should be fine.

In conclusion, I suggest updating `targetSdkVersion` of the app to use the same SDK in proj.android and proj.android-studio.

Comment (minggo) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java:71]: cocos2d-x supports Android 2.3+, so we can not invoke WebView .setWebContentsDebuggingEnabled. I think should invoke by reflection. Or it will cause compiling issue.

Comment (rynkowsg) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java:71]: Wait, cocos2d-x supports Android 2.3+ and it still will support. This is why I'm checking the system version at runtime (`Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT`).

We just need to keep minimumSdkVersion == 10, and use newer SDK to compile the code. In fact Cocos2d-x actually already use android-22 as a target platform, but only in the proj.android-studio projects.

Cocos2d-x supports two ways of compiling sources:
- old style with Ant (proj.android directories) and
- new style with Gradle (proj.android-studio directories).

In all proj.android-studio projects, cocos2d-x uses android-22 as targetSdkVersion. In all. Check this listening:

```
$ grep -rn "targetSdkVersion " .    
./cocos/platform/android/libcocos2dx/build/outputs/lint-results.html:136:<span class="lineno">   9</span>         <span class="errorspan">targetSdkVersion 22</span>
./cocos/platform/android/libcocos2dx/build/outputs/lint-results.xml:118:        errorLine1="        targetSdkVersion 22"
./cocos/platform/android/libcocos2dx/build.gradle:9:        targetSdkVersion 22
./cocos/platform/android/libcocos2dx-with-controller/build.gradle:9:        targetSdkVersion 22
./templates/cpp-template-default/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./templates/js-template-default/frameworks/runtime-src/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./templates/lua-template-default/frameworks/runtime-src/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/cpp-empty-test/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/cpp-tests/proj.android-studio/app/build/outputs/lint-results.html:45:<span class="lineno">  10</span>         <span class="errorspan">targetSdkVersion 22</span>
./tests/cpp-tests/proj.android-studio/app/build/outputs/lint-results.xml:16:        errorLine1="        targetSdkVersion 22"
./tests/cpp-tests/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/game-controller-test/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/js-memory-gc-tests/project/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/js-tests/project/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/lua-empty-test/project/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/lua-game-controller-test/project/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/lua-tests/project/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/performance-tests/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
```

What we should do is to set the same target for legacy build process. If we don't fix, because of the omission, cocos2d-x will be tied to android-10 functionality only.

Check it out:
- https://medium.com/google-developers/picking-your-compilesdkversion-minsdkversion-targetsdkversion-a098a0341ebd#.j30m6ahvr
- https://developer.android.com/training/basics/supporting-devices/platforms.html?utm_campaign=adp_series_sdkversion_010616&utm_source=medium&utm_medium=blog#version-codes

Comment (rynkowsg) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java:71]: There is no compiling issues, if CppTest is run from `tests/cpp-tests/proj.android-studio` directory (Jenkins had a build break during compilation of `tests/cpp-tests/proj.android`).

**Comment by rynkowsg** [comment]
@minggo: If you want I can prepare separate pull request to make both Android build process consistent.

File: cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java (hunk 1 above)
Comment (minggo) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java:71]: But [WebView.setWebContentsDebuggingEnabled()](https://developer.android.com/reference/android/webkit/WebView.html#setWebContentsDebuggingEnabled%28boolean%29) is available since API 19(Android OS 4.4). It will cause compiling error if build with Android OS lower than 4.4.

Comment (rynkowsg) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java:71]: Yes, it will cause a compilation error. In fact this PR causes two compilation errors. Both errors are related to the version of SDK used for compilation. I wrote about it above.

In conclusion, this PR shouldn't be merged until someone will bump the targetSdkVersion for Ant builds (Gradle builds are using targeSdkVersion == 22).

**Comment by rynkowsg** [comment]
I closed the pull request because it causes build break on Android. It can be re-open in the future when both Android build processes will be consistent - the same `targeSdkVersion` (`targeSdkVersion` should be updated for `proj.android` projects).
